ide/db.dataview/src/org/netbeans/modules/db/dataview/output/dataexport/DataViewTableDataExportFileChooser.java

```diff
@@ -0,0 +1,106 @@
+import org.openide.util.NbBundle;
+
+/**
+ *
+ * @author Periklis Ntanasis <pntanasis@gmail.com>
+ */
+@NbBundle.Messages({
+    "LBL_FILE_CHOOSER=Export Table Data",
+    "LBL_OVEWRITE_DIALOG=Confirm Data Export",
+    "MSG_OVEWRITE_DIALOG=File already exists.\nDo you want to overwrite it?",
+    "MSG_EXPORT_DATA=Export Data..."
+})
+public class DataViewTableDataExportFileChooser {
+
+    private static final List<DataExporter> EXPORTERS = Arrays.asList(
+            new CSVDataExporter(),
+            new TSVDataExporter(),
+            new XLSXDataExporter()
+    );
+    
+    private static File defaultDirectory;
+
+    public static void extractAsFile(final JTable table) {
+        final JFileChooser fc = initializeFileChooser();
+        int returnVal = fc.showDialog(null, Bundle.LBL_FILE_CHOOSER());
+        switch (returnVal) {
+            case JFileChooser.APPROVE_OPTION:
+                FileFilter filter = fc.getFileFilter();
+                DataExporter selectedExporter = EXPORTERS.stream()
+                        .filter(exporter -> exporter.getFileFilter() == filter)
+                        .findAny().orElseThrow(() -> new AssertionError("No matching file exporter filter found."));
+                final File file = checkFileExtension(fc.getSelectedFile(), selectedExporter);
+                if (checkFile(file)) {
+                    final String[] columnNames = DataExportUtils.getColumnNames(table);
+                    final Object[][] content = DataExportUtils.getTableContents(table);
+                    BaseProgressUtils.showProgressDialogAndRun(
+                            () -> selectedExporter.exportData(
+                                    columnNames,
+                                    content,
+                                    file),
+                            Bundle.MSG_EXPORT_DATA());
+                }
+                break;
+        }
+    }
+
+    private static boolean checkFile(File file) {
+        if (file.exists()) {
+            int a = JOptionPane.showConfirmDialog(
+                    null,
+                    Bundle.LBL_OVEWRITE_DIALOG(),
+                    Bundle.MSG_OVEWRITE_DIALOG(),
+                    JOptionPane.YES_NO_OPTION);
+            return a == JOptionPane.YES_OPTION;
+        }
+        return true;
+    }
+
+    private static File checkFileExtension(File file, DataExporter exporter) {
+        if (!exporter.handlesFileFormat(file)) {
+            return new File(file.getAbsolutePath() + "." + exporter.getDefaultFileExtension());
+        }
+        return file;
+    }
+    
+    private static JFileChooser initializeFileChooser() {
+        final JFileChooser fc = new JFileChooser();
+        fc.setAcceptAllFileFilterUsed(false);
+        EXPORTERS.forEach(exporter -> fc.addChoosableFileFilter(exporter.getFileFilter()));
+        if (defaultDirectory == null) {
+            defaultDirectory = fc.getCurrentDirectory();
+        }
```

Comment:
> Also, I have implemented the suggestion about storing and using the current directory and also explicitly setting the default file filter. To be honest if I understand correctly the default behavior should also be constant between invocations and always return the user's "default" directory. I guess that this could change in theory while the application is still running but it seems a bit extreme scenario. Anyway, I have made the change and any other suggestion is mote than welcome.

Lines 98-100 are unnecessary:

https://docs.oracle.com/javase/7/docs/api/javax/swing/JFileChooser.html#setCurrentDirectory(java.io.File)

What I meant is, that you should modify `extractAsFile` and there the switch case for `JFileChooser.APPROVE_OPTION`. If the user uses that, query the the current directory and store that. The file chooser on the second call will start in the directory the user selected last.

Reply:
> What I meant is, that you should modify `extractAsFile` and there the switch case for `JFileChooser.APPROVE_OPTION`. If the user uses that, query the the current directory and store that. The file chooser on the second call will start in the directory the user selected last.

I see, so the point is to always open the last used directory so the user won't have to navigate again there if he wants to save many files. OK, that makes sense.

I will look all the above probably over the weekend and get back to you. Thanks for the feedback!

Reply:
Sorry for the late response, I was a bit busier than expected. I have made the relevant change about the file chooser current directory.